net/udp/udp_conn.c

```diff
@@ -816,6 +816,10 @@ int udp_bind(FAR struct udp_conn_s *conn, FAR const struct sockaddr *addr)
     }
 #endif
 
+  /* Interrupts must be disabled while access the UDP connection list */
+
+  net_lock();
```

Comment:
why extend the scope of lock? I can't find the unpaired lock/unlock call

Reply:
I think we both looked at only local scope.